src/tools/rbd_mirror/CMakeLists.txt

```diff
@@ -21,7 +23,8 @@ set(rbd_mirror_internal
   image_sync/SnapshotCopyRequest.cc
   image_sync/SnapshotCreateRequest.cc
   image_sync/SyncPointCreateRequest.cc
-  image_sync/SyncPointPruneRequest.cc)
+  image_sync/SyncPointPruneRequest.cc
+  leader_watcher/Types.cc)
```

Comment:
Can you move this to a ```rbd_mirror_types``` internal library and add the new types to ceph_dencoder?